Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift

```diff
@@ -6,18 +6,23 @@ import TuistSupport
 // MARK: - Dependencies Graph Controller Errors
 
 enum DependenciesGraphControllerError: FatalError, Equatable {
-    case failedToEncodeDependeniesGraph
+    case failedToDecodeDependenciesGraph
+    case failedToEncodeDependenciesGraph
 
     var type: ErrorType {
         switch self {
-        case .failedToEncodeDependeniesGraph:
+        case .failedToDecodeDependenciesGraph:
+            return .abort
+        case .failedToEncodeDependenciesGraph:
             return .bug
         }
     }
 
     var description: String {
         switch self {
-        case .failedToEncodeDependeniesGraph:
+        case .failedToDecodeDependenciesGraph:
+            return "Couldn't decode the DependenciesGraph from the serialized JSON file. Running `tuist dependencies fetch` should solve the problem."
```

Comment:
Maybe we should automate this for the user? E.g. if decoding fails, we will try to regenerate the encoded graph and iff that fails, we throw an error (that will be of a type bug)

Reply:
I think it is a pretty uncommon scenario and I'm not sure it's worthy to optimize it. A much more relevant scenario would be to generate the graph if the `graph.json` is not present but the `Dependencies.swift` is.

In any case I think it makes sense to start simple in the first iteration and we can improve it later on (maybe we can add warning/TODO or open issues to keep track).

Reply:
I agree we can postpone solving this. Could you create an issue then?

Reply:
✅  https://github.com/tuist/tuist/projects/6#card-63929586